src/core/namespace.c

```diff
@@ -1303,6 +1401,32 @@ int setup_namespace(
                         return log_debug_errno(r, "Failed to decrypt dissected image: %m");
         }
 
+        if (n_mount_paths > 0) {
+                mount_path_loop_devices = new0(LoopDevice *, n_mount_paths);
+                if (!mount_path_loop_devices)
+                        return -ENOMEM;
+
+                for (i = 0; i < n_mount_paths; i++) {
+                        struct stat st;
+
+                        if (stat(mount_paths[i].source, &st) < 0)
+                                return -errno;
+
+                        if (S_ISBLK(st.st_mode)) /* Source is a block device */
+                                mount_path_loop_devices[i] = NULL;
+                        else {
+                                r = loop_device_make_by_path(mount_paths[i].source,
+                                                             mount_paths[i].read_only ? O_RDONLY : O_RDWR,
+                                                             0,
+                                                             &mount_path_loop_devices[i]);
+                                if (r < 0)
+                                        return log_debug_errno(r, "Failed to create loop device for a MountPath: %m");
+                        }
+                        if (streq(mount_paths[i].destination, "/"))
+                                root_mount_path = true;
```

Comment:
why have this? shouldn't this just generate an error?

Reply:
It's not an error to mount something at root, it can be used in place of `RootImage=`, but allow  any mount options (specific to file system or otherwise).

Reply:
not liking this i must say... I'd rather reduce the redundant options...

Reply:
There could be some redundancy, but each file system supports several unique options and it's not possible to use them now. There are also generic flags like nosuid (which is not enabled for RootImage).

Reply:
many file systems allow embedding mount options into these superblocks. I kinda wish we'd make more use of that, i.e. if an fs image is trusted sufficiently, we just apply the options from the superblocks.